src/MassTransit/SqlTransport/Configuration/AccessToken/SqlTransportPasswordRetriever.cs

```diff
@@ -0,0 +1,23 @@
+namespace MassTransit.AccessToken;
+
+using System;
+using Amazon;
+
+
+public static class SqlTransportPasswordRetriever
+{
+    public static string GetPassword(SqlTransportOptions options)
+    {
+        if (options.AuthenticationMode == AuthenticationMode.Password)
+            return options.Password;
+
+        var tokenGenerator = options.TokenIssuerOptions.Issuer switch
```

Comment:
Any sort of interface like this would have to be injected via configuration, with a default fallback to using the password from the transport options.